book/service_container.rst

```diff
@@ -744,15 +744,75 @@ Injecting the dependency by the setter method just needs a change of syntax:
     and "setter injection". The Symfony service container also supports
     "property injection".
 
-Making References optional
+Making References Optional
 --------------------------
 
 Sometimes, one of your services may have an optional dependency, meaning
 that the dependency is not required for your service to work properly. In
 the example above, the ``app.mailer`` service *must* exist, otherwise an exception
 will be thrown. By modifying the ``app.newsletter_manager`` service definition,
-you can make this reference optional. The container will then inject it if
-it exists and do nothing if it doesn't:
+you can make this reference optional, there are two strategies for doing this.
+
+Setting Missing Dependencies to null
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to explicitly set the argument to ``null`` in the event that the
+service does not exist then you may use the ``null`` strategy as follows:
+
+.. configuration-block::
+
```

Comment:
Its given just below the code examples :)

On Fri, Apr 29, 2016 at 06:09:01AM -0700, Oskar Stark wrote:
>    In [1]book/service_container.rst:
> 
>  >  --------------------------
>  >
>  >  Sometimes, one of your services may have an optional dependency, meaning
>  >  that the dependency is not required for your service to work properly. In
>  >  the example above, the ``app.mailer`` service *must* exist, otherwise an exception
>  >  will be thrown. By modifying the ``app.newsletter_manager`` service definition,
>  > -you can make this reference optional. The container will then inject it if
>  > -it exists and do nothing if it doesn't:
>  > +you can make this reference optional, there are two strategies for doing this.
>  > +
>  > +Setting Missing Dependencies to null
>  > +~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>  > +
>  > +If you want to explicitly set the argument to ``null`` in the event that the
>  > +service does not exist then you may use the ``null`` strategy as follows:
>  > +
>  > +.. configuration-block::
>  > +
> 
>    really? if yes, sorry, i didn't know that, but then this would be a good
>    information here ;-)
> 
>    what do you think?
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony/symfony-docs/pull/6511#discussion_r61573614
>    2. https://github.com/symfony/symfony-docs/pull/6511/files/0cf3bab63f400c6d04ff7baa1664c6f2b0f66d73#r61573614
>    3. https://github.com/symfony/symfony-docs/pull/6511/files/0cf3bab63f400c6d04ff7baa1664c6f2b0f66d73#r61573614